.github/workflows/gqcp.yml

```diff
@@ -0,0 +1,114 @@
+name: GQCP
+
+on: [pull_request]
+
+# These variables are for the slack notification, the default is failure, if all build steps are successful they will be updated to green and "Build successful"
+env:
+  BUILD_COLOR: '#FF0000'
+  BUILD_MESSAGE: 'Build failed'
```

Comment:
I'd like to propose the following changes:

```
COLOR_SUCCESSFUL = "#00FF00"  # green
COLOR_FAILED = "#FF000"  # red
MESSAGE_SUCCESSFUL = "Build successful"
MESSAGE_FAILED = "Build failed"

SLACK_COLOR = COLOR_FAILED
SLACK_MESSAGE = SLACK_MESSAGE_FAILED
```

Reply:
The slack environment only exists after "using" the action. 
`uses: rtCamp/action-slack-notify@master` However I want to update them these variables outside of the slack step, so I pre-initialize variables containing the information I want, update them (potentially) and then initialize the slack variables with those values later on.

Reply:
Okay, I understand what you mean here now. I don't know if the variables 'jobs.build.Slack%20Notifications.env.SLACK_MESSAGE` and `env.SLACK_MESSAGE` can both exist, we will see.